src/components/ProposalHero/ProposalHero.tsx

```diff
@@ -107,6 +114,83 @@ const VoteCTA: React.FC<VoteCTAProps> = ({
   )
 }
 
+type ExecutionDelayCTAProps = {
+  votingDeadlineBlock: number
+}
+
+const ExecutionDelayCTA: React.FC<ExecutionDelayCTAProps> = ({
+  votingDeadlineBlock
+}) => {
+  const { timeRemaining, targetBlock } = useExecutionDelayTimeRemaining(
+    votingDeadlineBlock
+  )
+
+  const { status: userStatus, user: accountUser } = useAccountUser()
+  const activeStake = accountUser
+    ? getActiveStake(accountUser)
+    : Utils.toBN('0')
+  const isUserStaker = userStatus === Status.Success && !activeStake.isZero()
+
+  return (
+    <div className={styles.voteCTA}>
+      <div className={styles.voteStatusContainer}>
+        {timeRemaining !== null && (
+          <>
+            <div className={styles.title}>{messages.executionDelay}</div>
+            <div className={styles.time}>
+              {getHumanReadableTime(timeRemaining)}
+            </div>
+          </>
+        )}
+        <div className={styles.blocks}>
+          <span>{`${messages.targetBlock}: ${targetBlock}`}</span>
+        </div>
+      </div>
+      {isUserStaker && (
+        <div>
+          <Button
+            text={messages.executeProposal}
+            type={ButtonType.GREEN}
+            isDepressed={true}
+            isDisabled={true}
+          />
+        </div>
+      )}
+    </div>
+  )
+}
+
+type ExecuteProposalCTAProps = {
+  onExecuteProposal: () => void
+}
+
+const ExecuteProposalCTA: React.FC<ExecuteProposalCTAProps> = ({
+  onExecuteProposal
+}) => {
+  const { status: userStatus, user: accountUser } = useAccountUser()
+  const activeStake = accountUser
+    ? getActiveStake(accountUser)
+    : Utils.toBN('0')
+  const isUserStaker = userStatus === Status.Success && !activeStake.isZero()
+
+  return (
+    <div className={styles.voteCTA}>
+      <div className={styles.voteStatusContainer}>
+        <div className={styles.title}>{messages.awaitExecuteProposal}</div>
+      </div>
+      {isUserStaker && (
```

Comment:
Do you have to be a staker to execute? i would think anyone could hypothetically do this